### DISCUSSION TITLE
Having Trouble with Homing Cycles

### DISCUSSION THREAD

**Discussion by Donagin** [terjeio/grblHAL#305] [Q&A]
Hi.  I am building a very unusual machine, and I would like to force homing in the Y and Z directions, but not in the X direction.  This is because my Y and Z are used to set the workpiece against different tools (pretty important to be homed), but the X has a 5 foot 2X6 that will not allow the machine to home without a fairly involved process.  I think I have seen allusions to individual axis homing, but am not sure where I saw this.  Apparently it will take more than $22 settings.  Can someone point me in the right direction?  I have individual axis home buttons on my custom sender I am writing (about half done), and am quite comfortable manipulating the code I write.

**Comment by phil-barrett** [comment]
Maybe $44?

**Reply by terjeio** [reply]
[$22-$27 and $43-$49](https://github.com/grblHAL/core/wiki/Additional-or-extended-settings) cover homing. I see now that the documentation is not complete, however up to date details are now available from the controller. E.g.` $$=22` issued from a terminal window outputs this information:

```
$22: Homing cycle as bitfield where setting bit 0 enables the rest:
    0 - Enable (1)
    1 - Enable single axis commands (2)
    2 - Homing on startup required (4)
    3 - Set machine origin to 0 (8)
    4 - Two switches shares one input pin (16)
    5 - Allow manual (32)
    6 - Override locks (64)
    7 - Keep homed status on reset (128)
ok
```

and `$help homing` this:
```
$22: Homing cycle as bitfield where setting bit 0 enables the rest:
    0 - Enable (1)
    1 - Enable single axis commands (2)
    2 - Homing on startup required (4)
    3 - Set machine origin to 0 (8)
    4 - Two switches shares one input pin (16)
    5 - Allow manual (32)
    6 - Override locks (64)
    7 - Keep homed status on reset (128)
$23: Homing direction invert as axismask
$24: Homing locate feed rate in mm/min
$25: Homing search seek rate in mm/min
$26: Homing switch debounce delay in milliseconds
$27: Homing switch pull-off distance in mm
$43: Homing passes, range: 1 - 128
$44: Axes homing, first pass as axismask
$45: Axes homing, second pass as axismask
$46: Axes homing, third pass as axismask
ok
```

If your custom sender has a UI for handlling settings then [check this wiki page](https://github.com/grblHAL/core/wiki/Report-extensions#code-sets-and-enumerations).

**Reply by Donagin** [reply]
Thanks so much for the replies.  I will use the $$=XX trick - that was something else I hadn't picked up on.  BTW Phil, my project is coming along nicely, and I continue to be happy with your board.  And Terjeio, I am very impressed and grateful for your software and work.  I don't think anyone will be interested in copying my work, because I am using this for such a specific task - but your work has given me incredibly powerful tools to make incredibly powerful tools.

-Don

![IMG_20210520_185300885](https://user-images.githubusercontent.com/84947078/119987347-8b655c80-bf8a-11eb-997c-9e74493307a9.jpg)
![IMG_20210526_182736219](https://user-images.githubusercontent.com/84947078/119987350-8bfdf300-bf8a-11eb-8389-b69c03f77365.jpg)
![IMG_20210520_185123024](https://user-images.githubusercontent.com/84947078/119987355-8bfdf300-bf8a-11eb-8518-fc45de0b0336.jpg)
![IMG_20210520_185149012](https://user-images.githubusercontent.com/84947078/119987356-8c968980-bf8a-11eb-8b5d-f95ceb6e400a.jpg)
